File: server/src/main/java/message/scmessage/roommanagement/ScJoinRoom.java
package message.scmessage.roommanagement;

import entity.user.User;
import message.scmessage.ScMessage;

public class ScJoinRoom extends ScMessage {
    public int statusCode;
    public User user;

    public ScJoinRoom(int statusCode) {
        this.statusCode = statusCode;
    }

    public ScJoinRoom(int statusCode, User user) {
        this.statusCode = statusCode;
        this.user = user;
    }
}

File: Client/Fruitster/Assets/Script/NetConnection/Messages/ScMessages/Room/ScJoinRoom.cs
using System;
using System.Net;
using Script.Model;
using UnityEngine;
using UnityEngine.SceneManagement;

namespace Script.Messages.ScMessages
{
    [Serializable]
    public class ScJoinRoom : ScMessage
    {
        public User user;
        public HttpStatusCode statusCode;
        public override void OnMessage(Session session)
        {
            if (statusCode == HttpStatusCode.OK)
            {
                if (user.userID ==UserProperties.MainPlayer.userID)
                {
                    SingletonDontDestroy.Instance.DoAction(() =>SceneManager.LoadScene("RoomScene"));
                }
                else
                {
                    UserProperties.UserRoom.Players.Add(user);
                    SingletonDontDestroy.Instance.DoAction(() => RoomSceneScript.Instance.RefreshRoom());
                }
                
            } else
            {
                Debug.Log(statusCode);
            }
        }
    }
}